src/librustc_typeck/check/method/probe.rs

```diff
@@ -799,7 +807,38 @@ impl<'a, 'gcx, 'tcx> ProbeContext<'a, 'gcx, 'tcx> {
             return Err(MethodError::PrivateMatch(def, out_of_scope_traits));
         }
 
+        debug!("Probing for method names similar to {:?}",
+               self.method_name);
+
+        let steps = self.steps.clone();
+        let lev_candidates = self.probe(|_| {
+            let mut pcx = ProbeContext::new(self.fcx, span, self.mode, self.method_name,
+                                            self.return_type, vec![]);
+            pcx.allow_similar_names = true;
+            pcx.steps = steps;
```

Comment:
I don't know if this might cause issues. The constructor for ProbeContext wants a Vec<CandidateStep>, but internally it's wrapped in an Rc.

Reply:
that shouldn't be a problem.

Reply:
Ok, done.